### PR TITLE
Add testcases of index type i32 for NonMaxSuppression

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Add template plugin testcases of i32 to op_reference/non_max_suppression.cpp*
 - *Remove skip regex from skip_tests_config.cpp*

### Tickets:
 - *70324*

### CODE REVIEW

**Comment by steve-y** [comment]
NonMaxSuppression-3,4,5 support index types of i32 and i64, but NonMaxSuppression-1 supports one index type of i32 or i64.
The tests in this PR supports i64 in NonMaxSuppression-1.

File: docs/template_plugin/tests/functional/op_reference/non_max_suppression.cpp [hunk 1]
```diff
@@ -1528,9 +1534,6 @@ std::vector<NonMaxSuppression1Params> generateParams1() {
 
 std::vector<NonMaxSuppression1Params> generateCombinedParams1() {
     const std::vector<std::vector<NonMaxSuppression1Params>> generatedParams {
-        generateParams1<element::Type_t::bf16, element::Type_t::i32, element::Type_t::f32, element::Type_t::i32>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/non_max_suppression.cpp:1531]: Why did you remove these test cases?

Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/non_max_suppression.cpp:1531]: Because NonMaxSuppression-3,4,5 has its type T_IND which is i32 or i64, but NonMaxSuppression1 does not specify the type and specify integer in its documentation, and the default type seems to be i64, so i32 tests in NonMaxSuppression-1 are removed instead skipping them in skip_tests_config.cpp.

**Review by ilyachur — APPROVED** [review]
